tree.c

```diff
@@ -668,9 +668,11 @@ static int special_album_cmp_date(const struct album *a, const struct album *b)
 	if (cmp)
 		return cmp;
 
-	cmp = a->date - b->date;
-	if (cmp)
-		return cmp;
+	if (!sort_albums_by_name) {
```

Comment:
I'd say this is questionable: function is called `special_album_cmp_date`. PR should aim to make the code better after adding the new feature, not for the minimal possible diff.

Reply:
True; this could be done in add_album, which already decides to do alphabetical sort for compilations, I think. Missed this while skimming the diff.

Reply:
I had actually done it properly in a previous draft. :facepalm:  I'm not sure what caused me to think this would be better, but originally I was calling the setting `sort-albums-by-date=0` so it stood out less then at least.